apps/http-sample/src/cats/cats.service.spec.ts

```diff
@@ -0,0 +1,236 @@
+            age: '10',
+            breed: 'Russian',
+            id: 1,
+          },
+          {
+            name: 'cat #2',
+            age: '5',
+            breed: 'Russian',
+            id: 2,
+          },
+        ],
+        headers: {},
+        config: { url: 'http://localhost:3000/mockUrl' },
+        status: 200,
+        statusText: 'OK',
+      } as any),
+    );
+
+    httpService.get('http://localhost:3000/mockUrl').subscribe((res) => {
+      expect(res).toEqual(response);
+    });
+  });
+
+  it('should return one cat', () => {
+    const data = {
+      name: 'cat #1',
+      age: '10',
+      breed: 'Russian',
+      id: 5,
+    };
+
+    const response: AxiosResponse<any> = {
+      data,
+      headers: {},
+      config: { url: 'http://localhost:3000/mockUrl/1' },
+      status: 200,
+      statusText: 'OK',
+    };
+
+    jest.spyOn(httpService, 'get').mockReturnValueOnce(
+      of({
+        data: {
+          name: 'cat #1',
+          age: '10',
+          breed: 'Russian',
+          id: 5,
+        },
+        headers: {},
+        config: { url: 'http://localhost:3000/mockUrl/1' },
+        status: 200,
+        statusText: 'OK',
+      }) as any,
+    );
+
+    httpService.get('http://localhost:3000/mockUrl/1').subscribe((res) => {
+      expect(res).toEqual(response);
+    });
+  });
+
+  it('should return a new cat', () => {
+    const data = {
+      name: 'cat #1',
+      age: 10,
+      breed: 'Russian',
+      id: 5,
+    };
+
+    let createCatDto: any = {
+      name: 'cat #1',
+      age: 10,
+      breed: 'Russian',
+    };
+
+    const response: AxiosResponse<any> = {
+      data,
+      headers: {},
+      config: { url: 'http://localhost:3000/mockUrl' },
+      status: 201,
+      statusText: 'OK',
+    };
+
+    jest.spyOn(httpService, 'post').mockImplementation(() => of(response));
+
+    service.create(createCatDto).subscribe({
+      next: (val) => {
+        createCatDto = val;
+      },
+      error: (err) => {
+        throw err;
+      },
+      complete: () => {
+        expect(data);
+      },
+    });
+  });
+
+  it('should return a cat update', () => {
+    const data = {
+      name: 'cat #1',
+      age: '10',
+      breed: 'Russian',
+      id: 5,
+    };
+
+    const response: AxiosResponse<any> = {
+      data,
+      headers: {},
+      config: { url: 'http://localhost:3000/mockUrl/5' },
+      status: 200,
+      statusText: 'OK',
+    };
+
+    jest.spyOn(httpService, 'put').mockImplementation(
+      () =>
+        of({
+          data: {
+            name: 'cat #1',
+            age: '10',
+            breed: 'Russian',
+            id: 5,
+          },
+          headers: {},
+          config: { url: 'http://localhost:3000/mockUrl/5' },
+          status: 200,
+          statusText: 'OK',
+        }) as any,
+    );
+
+    httpService
+      .put('http://localhost:3000/mockUrl/5', data)
+      .subscribe((res) => {
+        expect(res).toEqual(response);
+      });
```

Comment:
To test the service, it should be something like this:

```suggestion
  it('should return a cat update', () => {
    const data = {
      name: 'cat #1',
      age: '10',
      breed: 'Russian',
      id: 5,
    };

    const response: AxiosResponse<any> = {
      data,
      headers: {},
      config: { url: 'http://localhost:3000/mockUrl/5' },
      status: 200,
      statusText: 'OK',
    };

    jest.spyOn(httpService, 'put').mockImplementation(
      () =>
        of({
          data: {
            name: 'cat #1',
            age: '10',
            breed: 'Russian',
            id: 5,
          },
          headers: {},
          config: { url: 'http://localhost:3000/mockUrl/5' },
          status: 200,
          statusText: 'OK',
        }) as any,
    );

    service.update(5, {name: 'cat #1' }).subscribe(res => expect(res).toEqual(response.data))
  });
```

Reply:
Ok I try, then I'll let you know

Reply:
Resolved!